development/openstack-base-focal-wallaby/README.md

```diff
@@ -1,269 +1,371 @@
+**TESTING ONLY - This is a development bundle.** See the `stable` directory for
+the stable bundles.
+
+This `openstack-base` bundle deploys a base OpenStack cloud. Its major elements
+include:
+
+* Ubuntu 20.04 LTS (Focal)
+* OpenStack Wallaby
+* Ceph Octopus
+
+Cloud services consist of Compute, Network, Block Storage, Object Storage,
+Identity, Image, and Dashboard.
+
+> **Note**: Modifications will typically need to be made to this bundle for it
+  to work in your environment.
 
 ## Requirements
 
-This example bundle is designed to run on bare metal using Juju 2.x with [MAAS][] (Metal-as-a-Service); you will need to have setup a [MAAS][] deployment with a minimum of 3 physical servers prior to using this bundle.
+The bundle is primarily designed to work with [MAAS][maas] as a backing cloud
+for Juju. The MAAS cluster must already be deployed with at least three
+(ideally identical) physical servers as nodes, where each has the following
+minimum resources:
 
-Certain configuration options within the bundle may need to be adjusted prior to deployment to fit your particular set of hardware. For example, network device names and block device names can vary, and passwords should be yours.
+* 8 GiB memory
+* Enough CPU cores to support your workload
+* Two disks
+* Two cabled network interfaces
 
-For example, a section similar to this exists in the bundle.yaml file.  The third "column" are the values to set.  Some servers may not have eno2, they may have something like eth2 or some other network device name.  This needs to be adjusted prior to deployment.  The same principle holds for osd-devices.  The third column is a whitelist of devices to use for Ceph OSDs.  Adjust accordingly by editing bundle.yaml before deployment.
+The first disk is used for the node's operating system, and the second is for
+Ceph storage.
 
-```
-variables:
-  openstack-origin:    &openstack-origin     distro
-  data-port:           &data-port            br-ex:eno2
-  worker-multiplier:   &worker-multiplier    0.25
-  osd-devices:         &osd-devices          /dev/sdb /dev/vdb
-```
+The first network interface is used for communication between cloud services
+(East/West traffic), and the second is for network traffic between the cloud
+and all external networks (North/South traffic).
 
-Servers should have:
+> **Important**: The three MAAS nodes are needed for the actual OpenStack
+  cloud; they do not include the Juju controller. You actually need a minimum
+  of four nodes. The controller node however can be a smaller system (1 CPU and
+  4 GiB memory). Juju [constraints][juju-constraints-controller] can be used to
+  target this smaller system at controller-creation time.
```

Comment:
This is a bit confusing. Line 22 implies this. If we need more we should link out to documentation for deploying juju.

Reply:
I'll reword.